### DISCUSSION TITLE
Using FSRS to schedule topic recall of my application

### DISCUSSION THREAD

**Discussion by soasada** [open-spaced-repetition/free-spaced-repetition-scheduler#33] [Q&A]
Hello team,

First of all, I want to thank you for the work you’re doing with spaced repetition algorithms—it’s incredibly helpful.

I have an e-learning application where students take multiple-choice question tests with four possible answers, only one of which is correct.

The study content is divided into topics (we could think of each topic as a deck), and each topic has a set of questions (we could think of each question as a card).

In my app, there isn’t a button to rate a question. I had thought about rating each question as "good" if it’s answered correctly and as "again" if it’s answered incorrectly or left blank. And every time a user answer a question I get the last scheduling info calculated and apply the rating to it, and store it in database so that I will have a history of all answers and their scheduling information.

I would like to implement FSRS to:

- Suggest to my users when they should review a topic.
- Show them which topics are the most challenging for them.

Does it make sense to apply FSRS in this way, given that each question will always be rated as either "good" or "again"?

**Comment by Vilhelm-Ian** [comment]
> Hello team,
> 
> First of all, I want to thank you for the work you’re doing with spaced repetition algorithms—it’s incredibly helpful.
> 
> I have an e-learning application where students take multiple-choice question tests with four possible answers, only one of which is correct.
> 
> The study content is divided into topics (we could think of each topic as a deck), and each topic has a set of questions (we could think of each question as a card).
> 
> In my app, there isn’t a button to rate a question. I had thought about rating each question as "good" if it’s answered correctly and as "again" if it’s answered incorrectly or left blank. And every time a user answer a question I get the last scheduling info calculated and apply the rating to it, and store it in database so that I will have a history of all answers and their scheduling information.
> 
> I would like to implement FSRS to:
> 
> * Suggest to my users when they should review a topic.
> * Show them which topics are the most challenging for them.
> 
> Does it make sense to apply FSRS in this way, given that each question will always be rated as either "good" or "again"?

I don't think an ideal time for reviewing a topic exists. But you can make a screen where they can see for which topic they have most Difficulty for the cards

**Reply by soasada** [reply]
Maybe show them if they have questions to review for a specific topic? That way they will know if they have to review the topic or not.

Showing the difficulty by topic is one of the features I thought using FSRS would work.

**Reply by Vilhelm-Ian** [reply]
when you meant topic. I thought you had a situation where you meant soomething like they are studying biology and you want them that they have a hard time time genetics and they should re-reaed the topic or see what they don't usterdstand. instead of having to answer the questinos again

**Reply by soasada** [reply]
Let's use the biology as an example, imagine I have a topic called "Genetics I" and 102 questions about that topic. A student read the topic in cards in Anki app for example, and then come to my app and answers the 102 questions twice, then I have in database 204 answers each with a scheduling information calculated from the ratings "good" for correct and "again" for incorrect or left blank. I can say for example if the 60% of the answers were rated as "again" you should review the topic again (today, or near future) which means you go and re-read it and then come back and answer the 102 questions again. I was thinking in something like that.

**Reply by Vilhelm-Ian** [reply]
can't find it. There was a supermemo article that said you can't predict the optimal timee to passively learn something. While when it comes to answering the questions use the interval that predicts the desired retention level. I think just having a graph with average difficulty for subtopics is best. Because that will give the learner objective measure with which topic he is struggling the most.

**Reply by soasada** [reply]
I see, understood. Thanks for helping me, I will use it to offer a graph with the difficulty for my students.
